### ISSUE TITLE
Lint for `useradd` without `-l` flag

### ISSUE THREAD

**Issue by m-ildefons** [hadolint/hadolint#553]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

This is related to the follwing issue: https://github.com/moby/moby/issues/5419. As explaned there, docker can run into problems when using the command `useradd` without the `-l` option. This is because a large sparse logfile is created by that command and the overlay filesystem may have trouble dealing with that. The `-l` option disables the creation, thus avoiding the whole problem.

In my opinion, Hadolint should emit a warning in case the `-l` flag is not present, because
  * build times increase dramatically
  * image sizes increase dramatically

Given the age of the linked issue, it is unlikely, that this will be fixed anytime soon in upstream docker.

This behaviour is most noticeable when using large uids (>8 digits). This may be necessary in cases, where interaction with identity providers like FreeIPA is necessary.

I'll gladly look into writing a rule for this.

### Expected behavior
Hadolint should emit a warning.

### Actual behavior
No warning is emitted.

### Issue with Docker
To illustrate the necessity of having this warning, build an image with the following Dockerfile.
**Dockerfile** (if relevant)
```Dockerfile
FROM debian:buster
RUN useradd -u 123456789 runuser
```

```Shell
│ ~/tmp/breakage │ᐅ time docker build -t test .
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM debian:buster
 ---> ef05c61d5112
Step 2/2 : RUN useradd -u 123456789 runuser
 ---> Running in a73b006f4dc3
Removing intermediate container a73b006f4dc3
 ---> 7d5c4ae51acc
Successfully built 7d5c4ae51acc
Successfully tagged test:latest

real    10m36.483s
user    0m0.071s
sys     0m0.035s
│ ~/tmp/breakage │ᐅ docker images
REPOSITORY                                   TAG                        IMAGE ID       CREATED          SIZE
test                                         latest                     7d5c4ae51acc   2 minutes ago    40.1GB

```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**
`Haskell Dockerfile Linter v1.22.1-6-gbbfa5bb`

**Additional environment details (OS, stack version, etc.)**
CentOS 7, Docker version 20.10.3, build 48d30b5
